firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/AnalyticsDeferredComponents.java

```diff
@@ -0,0 +1,185 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics;
+
+import static com.google.firebase.crashlytics.FirebaseCrashlytics.APP_EXCEPTION_CALLBACK_TIMEOUT_MS;
+import static com.google.firebase.crashlytics.FirebaseCrashlytics.FIREBASE_CRASHLYTICS_ANALYTICS_ORIGIN;
+import static com.google.firebase.crashlytics.FirebaseCrashlytics.LEGACY_CRASH_ANALYTICS_ORIGIN;
+
+import androidx.annotation.GuardedBy;
+import androidx.annotation.NonNull;
+import com.google.firebase.analytics.connector.AnalyticsConnector;
+import com.google.firebase.annotations.DeferredApi;
+import com.google.firebase.crashlytics.internal.Logger;
+import com.google.firebase.crashlytics.internal.analytics.AnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.analytics.BlockingAnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.analytics.BreadcrumbAnalyticsEventReceiver;
+import com.google.firebase.crashlytics.internal.analytics.CrashlyticsOriginAnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.analytics.UnavailableAnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.breadcrumbs.BreadcrumbHandler;
+import com.google.firebase.crashlytics.internal.breadcrumbs.BreadcrumbSource;
+import com.google.firebase.crashlytics.internal.breadcrumbs.DisabledBreadcrumbSource;
+import com.google.firebase.inject.Deferred;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/** @hide */
+public class AnalyticsDeferredComponents {
+  private final Deferred<AnalyticsConnector> analyticsConnectorDeferred;
+  private volatile AnalyticsEventLogger analyticsEventLogger;
+  private volatile BreadcrumbSource breadcrumbSource;
+
+  @GuardedBy("this")
+  private List<BreadcrumbHandler> breadcrumbHandlerList;
+  
+  public AnalyticsDeferredComponents(Deferred<AnalyticsConnector> analyticsConnectorDeferred) {
+    this(
+        analyticsConnectorDeferred,
+        new DisabledBreadcrumbSource(),
+        new UnavailableAnalyticsEventLogger());
+  }
+
+  public AnalyticsDeferredComponents(
+      Deferred<AnalyticsConnector> analyticsConnectorDeferred,
+      @NonNull BreadcrumbSource breadcrumbSource,
+      @NonNull AnalyticsEventLogger analyticsEventLogger) {
+    this.analyticsConnectorDeferred = analyticsConnectorDeferred;
+    this.breadcrumbSource = breadcrumbSource;
+    this.breadcrumbHandlerList = new ArrayList<>();
+    this.analyticsEventLogger = analyticsEventLogger;
+    init();
+  }
+
+  public BreadcrumbSource getDeferredBreadcrumbSource() {
+    return breadcrumbHandler -> {
+      synchronized (this) {
+        breadcrumbHandlerList.add(breadcrumbHandler);
```

Comment:
Actually, it looks like this list will unconditionally grow even when/if analytics is available

Reply:
done